### DISCUSSION TITLE
Cannot create template from postgres due to pg_cron launcher and pg_net worker connections

### DISCUSSION THREAD

**Discussion by nkmnz** [supabase/supabase#30438] [Questions]
Hi, I want to create a database template from `postgres` in my development environment like this: 
```
CREATE DATABASE postgres_copy_$(date +%Y%m%d_%H%M%S) WITH TEMPLATE postgres OWNER postgres;
```

Therefore, I must close all other connections to the `postgres` database. Unfortunately, it seems to be impossible to get rid of two processes:
```
SELECT pid, usename, application_name, backend_type 
FROM pg_stat_activity 
WHERE datname = 'postgres'
  AND pid <> pg_backend_pid();
```
returns:
```
 pid  | usename  | application_name  |    backend_type     
------+----------+-------------------+---------------------
 3229 | postgres |                   | pg_net 0.8.0 worker
 3230 | postgres | pg_cron scheduler | pg_cron launcher
```

Whenever I terminate them, they reappear immediately. Any ideas?

**Comment by steve-chavez** [comment]
> Therefore, I must close all other connections to the `postgres` database

Otherwise it will cause an error like:

```
ERROR:  55006: source database "postgres" is being accessed by other users
DETAIL:  There are 7 other sessions using the database.
```

>  The principal limitation is that no other sessions can be connected to the source database while it is being copied. CREATE DATABASE will fail if any other connection exists when it starts; 
> From https://www.postgresql.org/docs/current/manage-ag-templatedbs.html#MANAGE-AG-TEMPLATEDBS


> Whenever I terminate them, they reappear immediately

Those are background workers, they will always reappear. Right now it's not supported to create databases based on the postgres template.

**Reply by nkmnz** [reply]
Thank you for the explanation!
Is there a way to (temporarily) disable those background workers in dev or self hosted instances? It would be really convenient for running tests of my application-level code if I could set the whole database from a template.

**Reply by steve-chavez** [reply]
dev and self-hosted yes, you should remove both pg_cron and pg_net from `shared_preload_libraries` and then restart.
